### PR TITLE
Add a flag to control the C++ jax.jit behavior.

### PR DESCRIPTION
**Pull request by jblespiau**

### CODE REVIEW

File: jax/api.py [hunk 1]
```diff
@@ -86,6 +86,10 @@
 FLAGS = flags.FLAGS
 flags.DEFINE_bool("jax_disable_jit", bool_env("JAX_DISABLE_JIT", False),
                   "Disable JIT compilation and just call original Python.")
+flags.DEFINE_bool("experimental_cpp_jit", False,
+                  "A temporary flag to opt-out of the C++ jax.jit fast path."
```
Comment (tomhennigan) [inline jax/api.py:90]: Nit: the flag actually enables the C++ fast path so I'd suggest rewording to "Experimental: Enables C++ fast path for jitted functions. Will be removed in a future release of JAX."

Comment (tomhennigan) [reply jax/api.py:90]: Also, can we have an environment variable to set the default value like `JAX_DISABLE_JIT`? This is useful for cases where folks are not using `jax.config.config_with_absl()`

**Review by mattjj — APPROVED** [review]
LGTM!
